pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -120,13 +120,24 @@ func (pl *NodeAffinity) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1
 		return framework.Queue, err
 	}
 	if isMatched {
-		logger.V(4).Info("node was created or updated, and matches with the pod's NodeAffinity", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
-		return framework.Queue, nil
+		if originalNode == nil || !pl.enableSchedulingQueueHint {
+			// The node was created or updated with QueueHints disabled (preCheck could have blocked some previous updates),
+			// so unblocking if only the new Pod is matching.
+			logger.V(4).Info("node was created or updated, and matches with the pod's NodeAffinity", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
```

Comment:
```suggestion
			logger.V(4).Info("node was created, and matches with the pod's NodeAffinity", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
```
Since we have already checked `originalNode == nil`, we can determine it's a node add event here?

Reply:
Not in the case when QHints were disabled. Actually I will split it into two separate checks to make the code more readable.